pkg/staticnetworkconfig/generator.go

```diff
@@ -121,7 +122,15 @@ func (s *StaticNetworkConfigGenerator) executeNMStatectl(ctx context.Context, ho
 	stdout, stderr, retCode := executer.ExecuteWithContext(ctx, "nmstatectl", "gc", f.Name())
 	if retCode != 0 {
 		s.log.Errorf("<nmstatectl gc> failed, errorCode %d, stderr %s, input yaml <%s>", retCode, stderr, hostYAML)
-		return "", fmt.Errorf("failed to execute nmstatectl gc, check the value provided for 'network_yaml': %s", hostYAML)
+		errMsg := stderr
+		errRegex := "Error: (?P<message>.*)"
+		re := regexp.MustCompile(errRegex)
+		matches := re.FindStringSubmatch(stderr)
+		msgIndex := re.SubexpIndex("message")
+		if len(matches) > msgIndex {
+			errMsg = matches[msgIndex]
+		}
```

Comment:
This is all rather complicated for error handling.
Can you give an example of the message we're matching against?

Also will the host index we have here be useful to the user? In the kubeapi, for example aren't nmstate yamls provided as individual resources? How would they figure out which one was index 3, for example?

Reply:
The type of message we would match against looks like this:
```
Traceback (most recent call last):
  File "/usr/bin/nmstatectl", line 33, in <module>
    sys.exit(load_entry_point('nmstate==1.2.1', 'console_scripts', 'nmstatectl')())
  File "/usr/lib/python3.10/site-packages/nmstatectl/nmstatectl.py", line 74, in main
    return args.func(args)
  File "/usr/lib/python3.10/site-packages/nmstatectl/nmstatectl.py", line 397, in run_gen_config
    libnmstate.generate_configurations(state), use_yaml=use_yaml
  File "/usr/lib/python3.10/site-packages/libnmstate/nmstate.py", line 419, in generate_configurations
    net_state = NetState(desire_state, gen_conf_mode=True)
  File "/usr/lib/python3.10/site-packages/libnmstate/net_state.py", line 55, in __init__
    self._route = RouteState(
  File "/usr/lib/python3.10/site-packages/libnmstate/route.py", line 204, in __init__
    self._merge_routes(des_route_state, ifaces)
  File "/usr/lib/python3.10/site-packages/libnmstate/route.py", line 223, in _merge_routes
    raise NmstateValueError(rt.invalid_reason)
libnmstate.error.NmstateValueError: Route {'table-id': 254, 'destination': '0.0.0.0/0', 'next-hop-address': '192.168.126.1', 'next-hop-interface': 'eth1'} next hop to unknown interface
```

So we would match that last line where it contains the actual error message
```
libnmstate.error.NmstateValueError: Route {'table-id': 254, 'destination': '0.0.0.0/0', 'next-hop-address': '192.168.126.1', 'next-hop-interface': 'eth1'} next hop to unknown interface
```

This is based on feedback received in https://issues.redhat.com/browse/MGMT-9574

The host seemed like it would be useful based on the feedback in the issue, but what do you think? Is it better to send back the host name instead?

Reply:
Is there any reason this can't just split on new line and take the last entry?
Maybe there's something we can do before to catch some simple validations like yaml format?

To be honest I think this is too much processing for an error string.

Reply:
Gotcha, splitting on a new line showed up a little strangely so I went ahead and just split on the `Error:` string instead of using the full regex. 

That's a good idea to add a yaml format validation. I'll look into adding that tomorrow! 

There are instances where it's not a yaml format error, though.